src/base/bittorrent/session.cpp

```diff
@@ -3929,6 +3929,37 @@ quint64 Session::getAlltimeUL() const
     return m_statistics->getAlltimeUL();
 }
 
+QVariantMap Session::getStats() const
+{
+    const quint64 atd = getAlltimeDL();
+    const quint64 atu = getAlltimeUL();
+
+    // num_peers is not reliable (adds up peers, which didn't even overcome tcp handshake)
+    quint32 peers = 0;
+    foreach (BitTorrent::TorrentHandle *const torrent, torrents())
+        peers += torrent->peersCount();
+
+    QVariantMap map;
```

Comment:
Maybe `QVariantHash`.

Reply:
Or maybe it would be better to define a struct instead of using a QVariant container here?
Using a struct can define type for each field, whereas one must lookup the correct type when using a QVariant container.

Reply:
Shoot, I actually intended to switch to a struct but completely forgot. Thanks for the suggestion; I'll make the necessary changes.